contents/java/Spring.md

```diff
@@ -0,0 +1,193 @@
+to manage the components that make up an application.
+    
+In the Spring framework, the IoC container is represented by the interface ApplicationContext. In order to assemble beans,
+the container uses configuration metadata, which can be in the form of XML configuration or annotations.
+
+Below is one way to manually instantiate a container:
+```
+    ApplicationContext context = new ClassPathXmlApplicationContext("applicationContext.xml");
+```
+   
+We can then perform dependency injection by using the external XML file.
+It makes our code loosely coupled and easier for testing. For example, we can have a simple Employee class
+containing two fields id and name with its setters and getters and a method to display these informations.
+```java
+    package com.company;  
+  
+    public class Employee {  
+       private int id;  
+       private String name;  
+       private String city;  
+  
+       public int getId() {  
+           return id;  
+       }  
+       public void setId(int id) {  
+           this.id = id;  
+       }  
+       public String getName() {  
+           return name;  
+       }  
+       public void setName(String name) {  
+           this.name = name;  
+       }  
+       void display(){  
+           System.out.println(id+" "+name);  
+       }
+    }
+```
+    
+And the corresponding `applicationContext.xml` file could be:
+```
+    <?xml version="1.0" encoding="UTF-8"?>  
+    <beans  
+    xmlns="http://www.springframework.org/schema/beans"  
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"  
+    xmlns:p="http://www.springframework.org/schema/p"  
+    xsi:schemaLocation="http://www.springframework.org/schema/beans  
+                http://www.springframework.org/schema/beans/spring-beans-3.0.xsd">  
+  
+    <bean id="obj" class="com.company.Employee">  
+    <property name="id">  
+    <value>20</value>  
+    </property>  
+    <property name="name">  
+    <value>Amy</value>  
+    </property>  
+    </bean>  
+    </beans>
+```
+We are providing the information into the bean by this file. The property element invokes the setter method. The value subelement of property will assign the specified value.
+   
+To better understand how this dependency injection work, we can have a `Test.java` looks like the following:
+```java
+    public class Test {  
+        public static void main(String[] args) {   
+            Resource r=new ClassPathResource("applicationContext.xml");  
+            BeanFactory factory=new XmlBeanFactory(r);  
+          
+            Employee e=(Employee)factory.getBean("obj");  
+            s.display();
+        }  
+    }  
+```
+The out put of the above code will be `20 Amy`.
+
+### Spring JdbcTemplate
+
+Spring JdbcTemplate is a powerful mechanism to connect to the database and execute SQL queries. It internally uses
+JDBC API, but eliminates a lot of problems of it.
```

Comment:
Grammar and tone nit.

'eliminates ... of it' → 'alleviates ... associated with it'